tutorials/data_generation_variant.ipynb

```diff
@@ -0,0 +1,572 @@
+    "- Target data, so in our case pathogenic versus benign labels, are in `data_raw/variant/variant_target_values.csv`.\n",
+    "- The final variant processed data will be saved in `data_processed/variant/` folder, which in turns contains a folder for residue-level data and another one for atomic-level data. More details about such different levels will come a few cells below."
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "`get_pdb_files_and_target_data` is an helper function used to retrieve the raw pdb files names, variants information and target values in a list from the CSV:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def get_pdb_files_and_target_data(data_path):\n",
+    "\tcsv_data = pd.read_csv(os.path.join(data_path, \"variant_target_values.csv\"))\n",
+    "\tpdb_files = glob.glob(os.path.join(data_path, \"pdb\", '*.ent'))\n",
+    "\tpdb_files.sort()\n",
+    "\tpdb_file_names = [os.path.basename(pdb_file) for pdb_file in pdb_files]\n",
+    "\tcsv_data_indexed = csv_data.set_index('pdb_file')\n",
+    "\tcsv_data_indexed = csv_data_indexed.loc[pdb_file_names]\n",
+    "\tres_numbers = csv_data_indexed.res_number.values.tolist()\n",
+    "\tres_wildtypes = csv_data_indexed.res_wildtype.values.tolist()\n",
+    "\tres_variants = csv_data_indexed.res_variant.values.tolist()\n",
+    "\ttargets = csv_data_indexed.target.values.tolist()\n",
+    "\tpdb_names = csv_data_indexed.index.values.tolist()\n",
+    "\tpdb_files = [data_path + \"/pdb/\" + pdb_name for pdb_name in pdb_names]\n",
+    "\treturn pdb_files, res_numbers, res_wildtypes, res_variants, targets\n",
+    "\n",
+    "pdb_files, res_numbers, res_wildtypes, res_variants, targets = get_pdb_files_and_target_data(data_path)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## `QueryCollection` and `Query` objects"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "For each missense variant, so for each data point, a query can be created and added to the `QueryCollection` object, to be processed later on. Different types of queries exist, based on the molecular resolution needed:\n",
+    "\n",
+    "- In a `SingleResidueVariantResidueQuery` each node represents one amino acid residue.\n",
+    "- In a `SingleResidueVariantAtomicQuery` each node represents one atom within the amino acid residues.\n",
+    "\n",
+    "A query takes as inputs:\n",
+    "\n",
+    "- A `.pdb` file, representing the protein structure containing the missense variant.\n",
+    "- The chain identifier of the variant residue.\n",
+    "- The positional number of the variant residue.\n",
+    "- The insertion code, ??\n",
```

Comment:
ask Gayatri or Li what this is?
it is also never tested (as not None), so maybe we should include a test for it at some point when we know what it is. We can then open a separate issue for that.

Reply:
The insertion code is used when two residues have the same numbering. The combination of residue numbering and insertion code defines the unique residue (https://www.wwpdb.org/documentation/file-format-content/format33/sect9.html#ATOM)

Reply:
The tutorial still shows a couple of question marks here. This should either be replaced by above comment or should be removed as it looks kind of iffy.

Reply:
Look at the updated commits, I edited it

Reply:
sorry, somehow it wasnt up to date